### ISSUE TITLE
Inside if/else/end, rubocop complains about missing guard clause instead of extraneous return

### ISSUE THREAD

**Issue by gregnavis** [bbatsov/rubocop#2903]
Rubocop behaves oddly on:

```ruby
def positive?(number)
  if number > 0
    return true
  else
    false
  end
end
```

## Expected behavior

Rubocop complains about unnecessary `return` statement.

## Actual behavior

Rubocop complains about missing guard clause.

## Steps to reproduce the problem

Save the ruby snippet above in `test.rb` and run `rubocop`. You'll see:

```
$ rubocop test.rb 
Inspecting 1 file
C

Offenses:

test.rb:2:3: C: Use a guard clause instead of wrapping the code inside a conditional expression.
  if number > 0
  ^^

1 file inspected, 1 offense detected
```

## RuboCop version

Include the output of `rubocop -V`:

```
$ bundle exec rubocop -V
0.37.2 (using Parser 2.3.0.6, running on ruby 2.3.0 x86_64-linux)
```

**Comment by segiddins** [comment]
RuboCop wants:

```ruby
def positive?(number)
  return true if number > 0
  false
end
```

**Comment by alexdowad** [comment]
This can be closed.

**Comment by gregnavis** [comment]
@segiddins That's exactly the problem. Please notice that `rubocop` is also fine after I remove the `return` statement.

What I'm saying is that I find it odd that

```ruby
def positive?(number)
  return true if number > 0
  false
end
```

over

```
def positive?(number)
  if number > 0
   true
  else
    false
  end
end
```

**Comment by alexdowad** [comment]
Why don't you just write:

```ruby
def positive?(number)
  number > 0
end
```

**Comment by gregnavis** [comment]
Numbers, comparison, `true` and `false` were just an example. Sorry for not making this clear.

In general, I think this:

```ruby
def some_method(argument)
  if condition?(argument)
    return operation1(argument)
  else
    operation2(argument)
  end
end
```

should be turned into

```ruby
def some_method(argument)
  if condition?(argument)
    operation1(argument)
  else
    operation2(argument)
  end
end
```

instead of

```ruby
def some_method(argument)
  return operation1(argument) if condition?(argument)
  operation2(argument)
end
```

**Comment by lumeet** [comment]
I think @gregnavis's point is valid. Using the default configuration, both of the above mentioned alternatives are equally good. It's just that `Style/RedundantReturn` doesn't currently look into if-branches. If it did, we'd have two offenses. I don't know which one of them would have precedence when auto-correcting, though.

**Comment by gregnavis** [comment]
Any thoughts? If you think that

```ruby
def some_method(argument)
  if condition?(argument)
    operation1(argument)
  else
    operation2(argument)
  end
end
```

should be preferred then I might be able to spend time on contributing a patch.

**Comment by mikegee** [comment]
> If you think that [if/else/end] should be preferred then I might be able to spend time on contributing a patch.

No, disable the `Style/GuardClause` cop in your Rubocop config.

**Comment by gregnavis** [comment]
@mikegee, I'm sorry for not being clear. My point was _being able to continue using `Style/GuardClause` as it's useful in many cases_. It simply isn't too smart when it sees:

```ruby
def some_method(argument)
  if condition?(argument)
    return operation1(argument)
  else
    operation2(argument)
  end
end
```
